src/garage/tf/models/base.py

```diff
@@ -339,9 +339,15 @@ def parameters(self, parameters):
         """
         variables = self._get_variables()
         for name, var in variables.items():
-            if name in parameters:
-                var.load(parameters[name])
-            else:
+            found = False
+            # param name without model name
+            param_name = name[name.find(self.name) + len(self.name) + 1:]
```

Comment:
what if the model name is not globally-unique? will this cause an error or incorrect behavior? i think we need to think about this really carefully.

Reply:
it only compares the variable inside `model1` and `model2`, and match their parameters with the part of the names that exclude the model name. It doesn't involve any other models.

Reply:
What if I have Foo/model1 and Bar/model1 in my session?

Reply:
If you do model1_foo.parameters = model1_bar.parameters, their parameters will be the same, which is the intended usage. In the new NPO, we can do old_policy.model.parameters = new_policy.model.parameters to update policy.